utils/update-catalog-hashes.ts

```diff
@@ -0,0 +1,93 @@
+import fs from "node:fs";
+
+import fastGlob from "fast-glob";
+import yaml from "yaml";
+
+import {maybeUpdateCatalogHash} from "./maybe-update-catalog-hash";
+
+const ROOT_PACKAGE_NAME = "perseus";
+
+type ChangesetConfig = {
+    ignore?: string[];
+};
+
+type PnpmWorkspace = {
+    catalogs: {
+        prodDeps?: Record<string, string>;
+        peerDeps?: Record<string, string>;
+        devDeps?: Record<string, string>;
+    };
+};
+
+/**
+ * Load the changeset configuration to determine which packages are unpublished.
+ */
+function loadChangesetConfig(): ChangesetConfig {
+    const configContent = fs.readFileSync(".changeset/config.json", "utf-8");
+    return JSON.parse(configContent);
+}
+
+/**
+ * Load the pnpm workspace configuration containing catalog dependencies.
+ */
+function loadPnpmWorkspace(): PnpmWorkspace {
+    const workspaceContent = fs.readFileSync("pnpm-workspace.yaml", "utf-8");
+    return yaml.parse(workspaceContent);
+}
+
+/**
+ * Find all package.json files in the workspace.
+ */
+function findAllPackageJsons(): string[] {
+    const packageJsonPaths = fastGlob.sync("**/package.json", {
+        absolute: true,
+        ignore: ["**/node_modules/**", "**/dist/**"],
+    });
```

Comment:
This works fine, but one option that would be easier to maintain (and not require maintaining an ignore list is to use `git` to figure out which files to work with. Only `package.json` files that are checked into `git` should be considered... so `git ls-files "package.json" "**/package.json"` would find all `package.json` files that are tracked... meaning we'd ignore everything in any `node_module/` or `dist/` folder automatically.